src/modules/mavlink/mavlink_main.cpp

```diff
@@ -2394,6 +2394,30 @@ Mavlink::task_main(int argc, char *argv[])
 			}
 		}
 
+		// For legacy gimbals using the mavlink gimbal v1 protocol, we need to send out commands.
+		// We don't care about acks for these though.
+		if (_gimbal_v1_command_sub.updated()) {
+			vehicle_command_s cmd;
+			_gimbal_v1_command_sub.copy(&cmd);
+
+			// FIXME: filter for target system/component
+
+			mavlink_command_long_t msg{};
+			msg.param1 = cmd.param1;
+			msg.param2 = cmd.param2;
+			msg.param3 = cmd.param3;
+			msg.param4 = cmd.param4;
+			msg.param5 = cmd.param5;
+			msg.param6 = cmd.param6;
+			msg.param7 = cmd.param7;
+			msg.command = cmd.command;
+			msg.target_system = cmd.target_system;
+			msg.target_component = cmd.target_component;
+			msg.confirmation = 0;
+
+			mavlink_msg_command_long_send_struct(get_channel(), &msg);
```

Comment:
Note that our mavlink send helpers will silently drop this if the buffer is full.

At a minimum you could check the tx buffer availability before actually copying and sending.

https://github.com/PX4/PX4-Autopilot/blob/8b2016b4ed763e82c2f71946b9bc268d3f36b120/src/modules/mavlink/streams/COMMAND_LONG.hpp#L66

Reply:
Right but if a command like this is missing it should not be a problem.

Reply:
That should be fine then. To be paranoid you could also do this after our regular stream update so that gimbal v1 mavlink output can't starve our normal streams on a mediocre link.

https://github.com/PX4/PX4-Autopilot/blob/e1f726a1ff4d1fe4222dd6a4f38fa1c986c5c61e/src/modules/mavlink/mavlink_main.cpp#L2436-L2452